docs/infra/module-dependencies.md

```diff
@@ -90,3 +90,129 @@ parent -- depends on --> config_value
 data.parent -- depends on --> config_value
 child -- depends on --> data.parent
 ```
+
+### Create "data" modules to reference a group of resources created in another module
+
+Often, a root module B will depend on a group of related resources created in another root module A. Moreover, the resources in A are typically created by calling a reusable module in [infra/modules](/infra/modules/) rather than directly in the root module. In this situation, rather than create a separate config for each dependency, create a data module D that B can call to reference R's resources, and group D and R together in a folder e.g. `/infra/modules/component/resources` and `/infra/modules/component/data`. In order to DRY up how related resources are named, create an static `interface` module that both the `resources` module and `data` module can reference. The `interface` module should typically have only one variable as input, the `name` of the group of resources, and the outputs should be the names of related resources in the group.
+
+```terraform
+# root module A
+
+module "config" {
+  ...
+}
+
+module "component" {
+  source = "../modules/component/resources"
+  name   = module.config.component_name
+}
+```
+
+```terraform
+# root module B
+
+module "config" {
+  ...
+}
+
+module "component" {
+  source = "../modules/component/data"
+  name   = module.config.component_name
+}
+```
+
+```terraform
+# resources module infra/modules/component/resources
+
+module "interface" {
```

Comment:
Here too I wonder about the UX of using a generic name. If I had low context here, I would probably confused as to whose interface is being called here. I would expect to see something like

```suggestion
module "component_interface" {
```

Reply:
I'm open to this, since interface is indeed somewhat generic. I also wonder if there's a better name for the module than "interface". After writing this doc I wonder if a more appropriate name for the module would be "related_resource_names" or "subresource_names" or "grouped_resource_names" or "module_resource_names" or just "resource_names" to indicate that it's a module that provides names for resources that are in the module.

What do you think?

Reply:
I think interface is a fine enough name for it. If I had to change it (we don't) then I might change it to something like API, which still has interface in the name 😆 

Anyways, if we aren't going to make it `compontent_interface`, then just `interface` is fine

Reply:
Ok I'll leave it as is for now. I did kind of like `resource_names` in a world where all the resource_names are defined in there, even ones we don't need data sources for. But that's purely theoretical so for now I'll leave as is.